### PR TITLE
DataViews list layout: scroll selected item to top on mount

### PR DESCRIPTION
**Pull request by oandregal**
## What?

This PR scrolls the selected item to the top on mounting the list layout.

## Why?

To give it more visibility.

## How?

Add a effect that only runs on mount.

### 1. Select an item that is near the top:

- Before

https://github.com/user-attachments/assets/d3ec37d3-04b6-4e2c-942c-31c0ae03444f

- After

https://github.com/user-attachments/assets/eb3c151a-f810-4a94-90ba-24c5a5716e6a

### 2. Select an item that is near the bottom:

- Before

https://github.com/user-attachments/assets/d1926d61-482c-4a40-abb3-17b4916cbc13

- After

https://github.com/user-attachments/assets/f4e7696f-ab5f-44b7-8691-7c04455cb28a

### CODE REVIEW

**Comment by oandregal** [comment]
For full disclosure: I've heard this from folks so wanted to give it a try, but I'm unconvinced this is worth doing. Sharing for gathering feedback.

**Comment by ntsekouras** [comment]
I think it's a nice small enhancement and I guess we could also do that for other layouts like `table`.

Regarding the implementation I guess we'd need to take into account multi selection, no?

File: packages/dataviews/src/dataviews-layouts/list/index.tsx [hunk 1]
```diff
@@ -150,6 +150,16 @@ function ListItem< Item >( {
 		setIsHovered( isHover );
 	};
 
+	// This effect is to scroll the item to the top of the list only on mount,
+	// so no dependencies declared.
+	useEffect( () => {
```
Comment (ntsekouras) [inline packages/dataviews/src/dataviews-layouts/list/index.tsx:155]: We have a similar effect below. Do we need this one?

Comment (oandregal) [reply packages/dataviews/src/dataviews-layouts/list/index.tsx:155]: The one below is executed any time selection changes and it's for little adjustments like this: 

https://github.com/user-attachments/assets/f3335341-73ff-4203-a08d-6fa5768224bd

We need a different one that will trigger "on mount", as a proxy for "switching to list view".

**Comment by jameskoster** [comment]
I think there's sense in ensuring the selected item is _visible_ when switching layouts. In my testing that already seems to be the case?

I'm not sure it needs to be at the top of the list, but invite more feedback on that detail.

**Comment by matt-west** [comment]
> 1. Select an item that is near the top:

This feels quite disruptive as the selected item would have already been visible. Scrolling it to be at the top of the list is disorienting.

Thanks for exploring this @oandregal, but I think the existing behaviour is superior.

**Comment by oandregal** [comment]
I ran this by other people and they agree scrolling to the top is worse than the current behaviour, so I'm closing this one.

**Pull request closed without merging**